src/elements/content-sidebar/activity-feed/annotation-thread/useAnnotationAPI.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+
+import React from 'react';
+import type { MessageDescriptor } from 'react-intl';
+import API from '../../../../api/APIFactory';
+import {
+    ERROR_CODE_DELETE_ANNOTATION,
+    ERROR_CODE_EDIT_ANNOTATION,
+    ERROR_CODE_FETCH_ANNOTATION,
+} from '../../../../constants';
+
+import type { Annotation, AnnotationPermission } from '../../../../common/types/annotations';
+import type { BoxItemPermission } from '../../../../common/types/core';
+import type { FeedItemStatus } from '../../../../common/types/feed';
+import type { ElementOrigin, ElementsXhrError } from '../../../../common/types/api';
+
+import commonMessages from '../../../common/messages';
+import messages from './messages';
+
+type Props = {
+    annotationId: string,
+    api: API,
+    errorCallback: (
+        error: ElementsXhrError | Error,
+        code: string,
+        contextInfo?: Object,
+        origin?: ElementOrigin,
+    ) => void,
+    fileId: string,
+    filePermissions: BoxItemPermission,
+};
+
+type UseAnnotationAPI = {
+    annotation?: Annotation,
+    error?: {
+        message?: MessageDescriptor,
+        title?: MessageDescriptor,
+    },
+    handleDelete: ({ id: string, permissions: AnnotationPermission }) => any,
+    handleEdit: (id: string, text: string, permissions: AnnotationPermission) => void,
+    handleStatusChange: (id: string, status: FeedItemStatus, permissions: AnnotationPermission) => void,
+    isLoading: boolean,
+};
+
+const useAnnotationAPI = ({ annotationId, api, fileId, filePermissions, errorCallback }: Props): UseAnnotationAPI => {
+    const [annotation, setAnnotation] = React.useState();
+    const [error, setError] = React.useState();
+    const [isLoading, setIsLoading] = React.useState(true);
+
+    const getAnnotationSuccess = (fetchedAnnotation: Annotation): void => {
+        setAnnotation(fetchedAnnotation);
+        setError(undefined);
+        setIsLoading(false);
+    };
+
+    const annotationSuccessCallback = (updatedAnnotation: Annotation): void => {
+        setAnnotation(prevAnnotation => ({
+            ...updatedAnnotation,
+            replies: prevAnnotation ? prevAnnotation.replies : [],
+            isPending: false,
+        }));
+    };
+
+    const annotationDeleteSuccessCallback = () => {
+        // will emit delete event
+    };
+
+    const createAnnotationErrorCallback = (code: string) => {
+        let message;
+        switch (code) {
+            case ERROR_CODE_FETCH_ANNOTATION:
+                message = messages.errorFetchAnnotation;
```

Comment:
Minor, but we could use an object declared outside the component to map errors to messages.